### PR TITLE
implement modular-views.macro

### LINKED ISSUE #59: export modular-views.macro
The current implementation exposes the internals of modular, and requires installing codegen.macro. This should probably be inside a module. Further, following https://github.com/jpmorganchase/modular/issues/49, it'll probably have some 'proprietary' logic, and will be hard to change across releases if we don't encapsulate it. 

Opening this issue as a stub, will decide an implementation after #49 lands.

### PR DESCRIPTION
**Pull request by threepointone**
This implements the view map as a babel macro, replacing `app/views.ts` from our previous iteration. This is written in plain javascript because it's otherwise hard to author and test the macro. In a future revision, I'll add jsdoc annotations
(https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html)

I also took this opportunity to delete docs/intro.md.

Fixes #59

### CODE REVIEW

File: packages/modular-views.macro/src/index.macro.js [hunk 1]
```diff
@@ -3,45 +3,51 @@ const path = require('path');
...
 
 function views({ references, babel }) {
-  // eslint-disable-next-line
-  const { types: t } = babel;
-  // TODO: add a react import
-  references.default.forEach((ref) => {
-    ref.replaceWithSourceString(
-      generateViewMap(path.join(getModularRoot(), 'packages')),
+  if (references.default.length > 1) {
+    console.warn(
```
Comment (threepointone) [inline packages/modular-views.macro/src/index.macro.js:37]: this should probably throw an error

**Comment by threepointone** [comment]
tests failing only on CI, will fix over the weekend.

File: packages/modular-views.macro/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  'view-2': React.lazy(() => import('view-2')),
+  'view-3': React.lazy(() => import('view-3')),
+  'view-4': React.lazy(() => import('view-4')),
+  // ...
+  'view-n': React.lazy(() => import('view-n')),
+}
+
+Like magic!
+*/
+```
```
Comment (matt-d-webb) [inline packages/modular-views.macro/README.md:46]: Nice!

**Review by matt-d-webb — APPROVED** [review]

**Comment by threepointone** [comment]
merging this, I'll release it tonight.

**Pull request merged**